webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/api/DataStructures.scala

```diff
@@ -80,6 +83,13 @@ case class ApiToken(value: String) extends AnyVal {
   // Avoid printing the value in logs, regardless of token type
   override def toString: String = s"[REDACTED ApiToken]"
 
+  // For cases we need to print a part of the plain token for debug.
+  // Show the first 4 chars: enough to disambiguate, and preserves 166 bits of randomness.
+  def exposeSecretBeginning: String = {
+    val acceptableToReveal = 4
+    value.take(acceptableToReveal) + '*'.toString() * (value.size - acceptableToReveal)
```

Comment:
You don't actually need to use the real length of * here, it might be already a valuable information. A fixed 6 * or whatever would be better.

Reply:
Yes, I wondered about it. As it is a Rudder token is should always be 32 chars long, and I wanted to avoid provided "false information" about the length that would hamper debugging. I'll try to find a suitable format.

Reply:
Updated with a bit different solution